src/delay.rs

```diff
@@ -0,0 +1,119 @@
+//! A delay driver based on SysTick
```

Comment:
```suggestion
//! A delay driver based on SysTick.
```

...etc for the other comments